ush/oceanice_nc2grib2.sh

```diff
@@ -0,0 +1,319 @@
+    -set_date "${current_cycle}" -set_ftime "${aperiod} hour ave fcst" \
+    -set_scaling same same -set_grib_type c1 -grib_out "${outfile}" \
+  -import_netcdf "${infile}" "LwLatSens" "0:1:${latlon_dims}" \
+    -set_var THFLX -set center 7 \
+    -set_date "${current_cycle}" -set_ftime "${aperiod} hour ave fcst" \
+    -set_scaling same same -set_grib_type c1 -grib_out "${outfile}" \
+  -import_netcdf "${infile}" "MLD_003" "0:1:${latlon_dims}" \
+    -set_var WDEPTH -set center 7 -set_lev "mixed layer depth" \
+    -set_date "${current_cycle}" -set_ftime "${aperiod} hour ave fcst" \
+    -set_scaling same same -set_grib_type c1 -grib_out "${outfile}"
+
+# Additional variables needed for GFSv17/GEFSv13 operational forecast
+# files, but GRIB2 parameters not available in NCEP (-set center 7)
+# tables in wgrib2 v2.0.8:
+#
+#  -import_netcdf "${infile}" "Heat_PmE" "0:1:${latlon_dims}" \
+#    -set_var DWHFLUX -set center 7 \
+#    -set_date "${current_cycle}" -set_ftime "${aperiod} hour ave fcst" \
+#    -set_scaling same same -set_grib_type c1 -grib_out "${outfile}" \
+#  -import_netcdf "${infile}" "taux" "0:1:${latlon_dims}" \
+#    -set_var XCOMPSS -set center 7 \
+#    -set_date "${current_cycle}" -set_ftime "${aperiod} hour ave fcst" \
+#    -set_scaling same same -set_grib_type c1 -grib_out "${outfile}" \
+#  -import_netcdf "${infile}" "tauy" "0:1:${latlon_dims}" \
+#    -set_var YCOMPSS -set center 7 \
+#    -set_date "${current_cycle}" -set_ftime "${aperiod} hour ave fcst" \
+#    -set_scaling same same -set_grib_type c1 -grib_out "${outfile}"
+
+  rc=$?
+  # Check if the conversion was successful
+  if (( rc != 0 )); then
+   echo "FATAL ERROR: Failed to convert the ocean rectilinear netCDF file to grib2 format"
+  fi
+  return "${rc}"
+
+}
+
+################################################################################
+function _ocean3D_nc2grib2 {
+# This function converts the ocean 3D rectilinear netCDF files to grib2 format
+
+  # Set the inputs
+  local grid=${1} # 0p25, 0p50, 1p00, 5p00
+  local latlon_dims=${2} # 0:721:0:1440, 0:361:0:720, 0:181:0:360, 0:36:0:72
+  local levels=${3} # 5:15:25:35:45:55:65:75:85:95:105:115:125
+  local current_cycle=${4} # YYYYMMDDHH
+  local aperiod=${5} # 0-6
+  local infile=${6} # ocean.0p25.nc
+  local outfile=${7} # ocean_3D.0p25.grib2
+  local template=${8} # template.global.0p25.gb2
+
+  IFS=':' read -ra depths <<< "${levels}"
+
+  zl=0
+  for depth in "${depths[@]}"; do
+
+    [[ -f "tmp.gb2" ]] && rm -f "tmp.gb2"
+
+    ${WGRIB2} "${template}" \
+    -import_netcdf "${infile}" "temp" "0:1:${zl}:1:${latlon_dims}" \
+      -set_var WTMP -set center 7 -rpn "273.15:+" \
+      -set_lev "${depth} m below water surface" \
+      -set_date "${current_cycle}" -set_ftime "${aperiod} hour ave fcst" \
+      -set_scaling same same -set_grib_type c1 -grib_out tmp.gb2 \
+    -import_netcdf "${infile}" "so" "0:1:${zl}:1:${latlon_dims}" \
+      -set_var SALTY -set center 7 -rpn "1000.0:/" \
```

Comment:
Change `-set_var SALTY -set center 7 -rpn "1000.0:/ \` to
`-set_var SALIN -set center 7 \`